crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1298,7 +1301,9 @@ impl PgManager {
         coin_type: Option<String>,
     ) -> Result<Option<Balance>, Error> {
         let address = address.into_vec();
-        let coin_type = coin_type.unwrap_or("0x2::sui::SUI".to_string());
+        let coin_type = parse_to_type_tag(coin_type)
+            .map_err(|_| Error::Internal("Coin type is not valid".to_string()))?
```

Comment:
I think this would typically be a client error rather than an internal error

Reply:
There's a deeper problem (not addressable in this PR) with this interface that the data layer often won't know which kind of error it is, so really it needs its own error type, which the GraphQL code can use and reinterpret into internal or user errors based on context (This is why `PackageCache` defines its own error type).

Reply:
Actually before this change, we would try to find in the db objects whose type is equal to the provided `coin_type` string so if the coin type is invalid, we simply wouldn't find anything in the db so we would simply return `None`. Maybe we should keep doing that here? But having a new category of error for user errors and returning something like `InvalidInput("Coin type invalid")` also makes sense.

Reply:
Right, but whether that's an internal error, a user error, or allowed depends on the context in which the DB call is made.  It could be an internal error if we hardcoded a type, like `0x2::sui::SUI` and that turned out to be wrong (e.g. because we're now canonicalising everywhere).  Depending on the query, it could be a user error, because they gave a malformed type, or it could be allowed, because there simply are no coins of that type.  That's why it's useful to have a different error type at this level, so the caller can decide which case we're in.